projects/plugins/social/src/class-rest-settings-controller.php

```diff
@@ -26,24 +27,28 @@ class REST_Settings_Controller extends WP_REST_Controller {
 	 * @static
 	 */
 	public function register_rest_routes() {
-		register_rest_route(
-			'jetpack/v4',
-			'/social/settings',
-			array(
+		// If the site has an older version of Jetpack we still need to register the route.
+		if ( ! Publicize_Utils::has_new_module_endpoint() ) {
```

Comment:
This is very much a nitpick, so don't feel obliged, but we could reduce the nesting here by swapping the order these two endpoints are registered, and flipping this logic to return early

```suggestion
		if ( Publicize_Utils::has_new_module_endpoint() ) {
			return;
		}
```

Reply:
Sure, changed in 0df6110